libtxref/txref.cpp

```diff
@@ -220,6 +222,9 @@ namespace {
         if(isLengthValid(txref.length()) && (txref.at(0) == 'x' || txref.at(0) == '8')) {
             return std::string(txref::BECH32_HRP_TEST) + bech32::separator + txref;
         }
+        if(isLengthValid(txref.length()) && (txref.at(0) == 'q' || txref.at(0) == 'p')) {
```

Comment:
We could remove these heuristics if we created separate types for txrefs with and without HRPs.  Let's put that on the TODO list.